models/DataObject/ClassDefinition/Data/InputQuantityValue.php

```diff
@@ -96,13 +96,11 @@ public function getDataFromResource($data, $object = null, $params = [])
     public function getDataFromEditmode($data, $object = null, $params = [])
     {
         if ($data['value'] || $data['unit']) {
-            if ($data['unit']) {
-                if ($data['unit'] == -1 || $data['unit'] == null || empty($data['unit'])) {
-                    return $this->getNewDataObject($data['value'], null);
-                }
-
-                return $this->getNewDataObject($data['value'], $data['unit']);
+            if ($data['unit'] == -1 || empty($data['unit'])) {
```

Comment:
```suggestion
            if (empty($data['unit']) || $data['unit'] == -1) {
```
we should check for empty first to avoid undefined error.